caddyhttp/push/link_parser.go

```diff
@@ -0,0 +1,70 @@
+package push
+
+import (
+	"regexp"
+	"strings"
+)
+
+const (
+	commaSeparator     = ","
+	semicolonSeparator = ";"
+	equalSeparator     = "="
+)
+
+var (
+	resourceRegexp = regexp.MustCompile("^<(.*?)>;?(.*)$")
```

Comment:
How well does this work if resource URIs have commas or semicolons in them?

Also -- this isn't a showstopper -- I wonder how this pre-compiled regex performs compared to strings.Split. I'll try to whip up a benchmark today and see, just to satisfy my curiosity.